core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/WorkflowScheduler.scala

```diff
@@ -145,18 +144,11 @@ class WorkflowScheduler(
       akkaActorRefMappingService: AkkaActorRefMappingService,
       akkaActorService: AkkaActorService
   ): Unit = {
-    val builtOpsInRegion = new mutable.HashSet[LayerIdentity]()
-    var frontier: Iterable[LayerIdentity] = workflow.getSourcesOfRegion(region)
+    val builtOpsInRegion = new mutable.HashSet[PhysicalOpIdentity]()
+    var frontier: Iterable[PhysicalOpIdentity] = workflow.getSourcePhysicalOpsOfRegion(region)
     while (frontier.nonEmpty) {
-      frontier.foreach { (op: LayerIdentity) =>
-        val prev: Array[(LayerIdentity, OpExecConfig)] =
-          workflow.physicalPlan
-            .getUpstream(op)
-            .filter(upStreamOp =>
-              builtOperators.contains(upStreamOp) && region.getOperators.contains(upStreamOp)
-            )
-            .map(upStreamOp => (upStreamOp, workflow.getOpExecConfig(upStreamOp)))
-            .toArray // Last layer of upstream operators in the same region.
+      frontier.foreach { (op: PhysicalOpIdentity) =>
+        // Last PhysicalOp of upstream operators in the same region.
```

Comment:
remove this comment? it appears to be from an unused code fragment.

Reply:
removed